client/state/jetpack-connect/test/selectors.js

```diff
@@ -112,6 +112,51 @@ describe( 'selectors', () => {
 			expect( getFlowType( state, { slug: 'sitetest' } ) ).to.be.false;
 		} );
 	} );
+
+	describe( '#getJetpackSiteByUrl()', () => {
+		it( 'should return null if site is not found', () => {
+			const state = {
+				sites: {
+					items: {}
+				}
+			};
+
+			expect( getJetpackSiteByUrl( state, 'example.wordpress.com' ) ).to.be.null;
+		} );
+
+		it( 'should return false if the site is not a jetpack site', () => {
+			const state = {
+				sites: {
+					items: {
+						12345678: {
+							ID: 12345678,
+							URL: 'https://example.wordpress.com/',
+							jetpack: false
+						}
+					}
+				}
+			};
+
+			expect( getJetpackSiteByUrl( state, 'https://example.wordpress.com/' ) ).to.be.null;
+		} );
+
+		it( 'should return the site object if the site is a jetpack site', () => {
+			const state = {
+				sites: {
+					items: {
+						12345678: {
+							ID: 12345678,
+							URL: 'https://example.wordpress.com/',
+							jetpack: true
+						}
+					}
+				}
+			};
+
+			expect( getJetpackSiteByUrl( state, 'https://example.wordpress.com/' ) ).to.eql( state.sites.items[ 12345678 ] );
+		} );
+	} );
+
```

Comment:
I think you should add a test case where two the urls between .com site and jetpack site match. 
Something like
```
items: {
666: {
   ID: 666,
  URL: 'https://potato.com/',
 jetpack: false
 },
12345678: {
   ID: 12345678,
  URL: 'https://potato.com/',
 jetpack: true
 },
 }
``` This case does happen sometimes